modules/efk-logging-fluentd-undefined.adoc

```diff
@@ -0,0 +1,146 @@
+
+You can configure how cluster logging treats fields from disparate sources by editing the log collector daemonset, Fluentd or Rsyslog, and setting environment variables in the table below. 
+
+// from https://github.com/ViaQ/fluent-plugin-viaq_data_model/commit/8b5ef11cedec4c372b2cb082afc7f9cc08473654
+
+* Undefined fields. One of the problems with log data from disparate systems is that some fields might be unknown to the ViaQ data model.
+Such fields are called _undefined_. ViaQ requires all top-level fields to be defined and described. 
++
+Use the parameters to configure how {product-title} moves any undefined fields under a top-level field called `undefined` to avoid conflicting with the _well known_ ViaQ 
+top-level fields. You can add undefined fields to the top-level fields and move others to an `undefined` container.
++
+You can also replace special characters in undefined fields and convert undefined fields to their JSON string representation. 
+Coverting to JSON string preserves the structure of the value, so that you can retrieve the value later and convert it back to a map or an array.
++
+** Simple scalar values like numbers and booleans are changed to a quoted string. For example: `10` becomes `"10"`, `3.1415` becomes `"3.1415"``, `false` becomes `"false"`.
++
+** Map/dict values and array values are converted to their JSON string representation: `"mapfield":{"key":"value"}` becomes `"mapfield":"{\"key\":\"value\"}"` and `"arrayfield":[1,2,"three"]` becomes `"arrayfield":"[1,2,\"three\"]"`.
+
+* Defined fields. You can also configure which defined fields appear in the top levels of the logs. 
++
+[[default-fields]]
+The default top-level fields, defined through the `CDM_DEFAULT_KEEP_FIELDS` parameter, are `CEE`, `time`, `@timestamp`, `aushape`, `ci_job`, `collectd`, `docker`, `fedora-ci`, 
+`file`, `foreman`, `geoip`, `hostname`, `ipaddr4`, `ipaddr6`, `kubernetes`, `level`, `message`, `namespace_name`, `namespace_uuid`,
+`offset`, `openstack`, `ovirt`, `pid`, `pipeline_metadata`, `rsyslog`, `service`, `systemd`, `tags`, `testcase`, `tlog`, `viaq_msg_id`.
++
+Any fields not included in `${CDM_DEFAULT_KEEP_FIELDS}` or `${CDM_EXTRA_KEEP_FIELDS}` are moved to `${CDM_UNDEFINED_NAME}` if `CDM_USE_UNDEFINED` is `true`.
++
+[NOTE]
+====
+The `CDM_DEFAULT_KEEP_FIELDS` parameter is for only advanced users, or if you are instructed to do so by Red Hat support.
+==== 
+
+* Empty fields. You can determine which empty fields to retain from disparate logs.
+
+.Environment parameters for log normalization
+[cols="2,4,2", options="header"]
+|===
+|Parameters  |Definition  |Example
+
+|`CDM_EXTRA_KEEP_FIELDS`
+|Specify an extra set of defined fields to be kept at the top level of the logs in addition to the `CDM_DEFAULT_KEEP_FIELDS`. The default is "".
+|`CDM_EXTRA_KEEP_FIELDS="broker"`
+
+|`CDM_KEEP_EMPTY_FIELDS`
+|Specify fields to retain even if empty in CSV format. Empty defined fields not specified are dropped. The default is "".
```

Comment:
@mburke5678 We have a default value `message`, see https://github.com/openshift/cluster-logging-operator/blob/master/pkg/k8shandler/fluentd.go#L239 and https://github.com/openshift/cluster-logging-operator/blob/master/pkg/k8shandler/rsyslog.go#L366